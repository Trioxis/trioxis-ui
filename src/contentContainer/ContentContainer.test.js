import React from 'react';
import { mount } from 'enzyme';
import {ContentContainer} from './';

jest.mock('react-jss',()=>{
  jest.spyOn(global.Math,'random').mockImplementation(()=>0.42);
  return jest.requireActual('react-jss')
})

test('ContentContainer should render as snapshot defines', () => {
  const tree = mount(<ContentContainer />)
  expect(tree).toMatchSnapshot()
});
