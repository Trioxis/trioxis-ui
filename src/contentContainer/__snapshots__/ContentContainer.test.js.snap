// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentContainer should render as snapshot defines 1`] = `
<Jss(_ContentContainer)>
  <_ContentContainer
    classes={
      Object {
        "app": "app-0-0",
      }
    }
    sheet={
      StyleSheet {
        "0.42": null,
        "attached": true,
        "classes": Object {
          "app": "app-0-0",
        },
        "deployed": true,
        "linked": false,
        "options": Object {
          "Renderer": [Function],
          "classes": Object {
            "app": "app-0-0",
          },
          "generateClassName": [Function],
          "index": -100000,
          "insertionPoint": undefined,
          "jss": Jss {
            "generateClassName": [Function],
            "options": Object {
              "Renderer": [Function],
              "createGenerateClassName": [Function],
              "plugins": Array [
                Object {
                  "onCreateRule": [Function],
                  "onProcessRule": [Function],
                },
                Object {
                  "onProcessStyle": [Function],
                },
                Object {
                  "onProcessStyle": [Function],
                },
                Object {
                  "onProcessStyle": [Function],
                },
                Object {
                  "onProcessStyle": [Function],
                },
                Object {
                  "onChangeValue": [Function],
                  "onProcessStyle": [Function],
                },
                Object {
                  "onProcessStyle": [Function],
                },
                Object {
                  "onChangeValue": [Function],
                  "onProcessRule": [Function],
                  "onProcessStyle": [Function],
                },
                Object {
                  "onProcessStyle": [Function],
                },
              ],
            },
            "plugins": PluginsRegistry {
              "hooks": Object {
                "onChangeValue": Array [
                  [Function],
                  [Function],
                ],
                "onCreateRule": Array [
                  [Function],
                  [Function],
                  [Function],
                  [Function],
                  [Function],
                  [Function],
                  [Function],
                  [Function],
                  [Function],
                  [Function],
                ],
                "onProcessRule": Array [
                  [Function],
                  [Function],
                ],
                "onProcessSheet": Array [],
                "onProcessStyle": Array [
                  [Function],
                  [Function],
                  [Function],
                  [Function],
                  [Function],
                  [Function],
                  [Function],
                  [Function],
                ],
              },
            },
            "version": "8.1.0",
          },
          "meta": "Jss(_ContentContainer), Unthemed, Static",
          "parent": [Circular],
          "sheet": [Circular],
        },
        "renderer": DomRenderer {
          "element": <style
            data-jss=""
            data-meta="Jss(_ContentContainer), Unthemed, Static"
            type="text/css"
          >
            
.app-0-0 {
  margin: auto;
  max-width: 1200px;
}

          </style>,
          "getSelector": [Function],
          "getStyle": [Function],
          "hasInsertedRules": false,
          "setSelector": [Function],
          "setStyle": [Function],
          "sheet": [Circular],
        },
        "rules": RuleList {
          "classes": Object {
            "app": "app-0-0",
          },
          "index": Array [
            Object {
              "margin": "auto",
              "max-width": "1200px",
            },
          ],
          "map": Object {
            ".app-0-0": Object {
              "margin": "auto",
              "max-width": "1200px",
            },
            "app": Object {
              "margin": "auto",
              "max-width": "1200px",
            },
          },
          "options": Object {
            "Renderer": [Function],
            "classes": Object {
              "app": "app-0-0",
            },
            "generateClassName": [Function],
            "index": -100000,
            "insertionPoint": undefined,
            "jss": Jss {
              "generateClassName": [Function],
              "options": Object {
                "Renderer": [Function],
                "createGenerateClassName": [Function],
                "plugins": Array [
                  Object {
                    "onCreateRule": [Function],
                    "onProcessRule": [Function],
                  },
                  Object {
                    "onProcessStyle": [Function],
                  },
                  Object {
                    "onProcessStyle": [Function],
                  },
                  Object {
                    "onProcessStyle": [Function],
                  },
                  Object {
                    "onProcessStyle": [Function],
                  },
                  Object {
                    "onChangeValue": [Function],
                    "onProcessStyle": [Function],
                  },
                  Object {
                    "onProcessStyle": [Function],
                  },
                  Object {
                    "onChangeValue": [Function],
                    "onProcessRule": [Function],
                    "onProcessStyle": [Function],
                  },
                  Object {
                    "onProcessStyle": [Function],
                  },
                ],
              },
              "plugins": PluginsRegistry {
                "hooks": Object {
                  "onChangeValue": Array [
                    [Function],
                    [Function],
                  ],
                  "onCreateRule": Array [
                    [Function],
                    [Function],
                    [Function],
                    [Function],
                    [Function],
                    [Function],
                    [Function],
                    [Function],
                    [Function],
                    [Function],
                  ],
                  "onProcessRule": Array [
                    [Function],
                    [Function],
                  ],
                  "onProcessSheet": Array [],
                  "onProcessStyle": Array [
                    [Function],
                    [Function],
                    [Function],
                    [Function],
                    [Function],
                    [Function],
                    [Function],
                    [Function],
                  ],
                },
              },
              "version": "8.1.0",
            },
            "meta": "Jss(_ContentContainer), Unthemed, Static",
            "parent": [Circular],
            "sheet": [Circular],
          },
          "raw": Object {
            "app": Object {
              "margin": "auto",
              "maxWidth": "1200px",
            },
          },
        },
      }
    }
  >
    <div
      className="app-0-0"
      sheet={
        StyleSheet {
          "0.42": null,
          "attached": true,
          "classes": Object {
            "app": "app-0-0",
          },
          "deployed": true,
          "linked": false,
          "options": Object {
            "Renderer": [Function],
            "classes": Object {
              "app": "app-0-0",
            },
            "generateClassName": [Function],
            "index": -100000,
            "insertionPoint": undefined,
            "jss": Jss {
              "generateClassName": [Function],
              "options": Object {
                "Renderer": [Function],
                "createGenerateClassName": [Function],
                "plugins": Array [
                  Object {
                    "onCreateRule": [Function],
                    "onProcessRule": [Function],
                  },
                  Object {
                    "onProcessStyle": [Function],
                  },
                  Object {
                    "onProcessStyle": [Function],
                  },
                  Object {
                    "onProcessStyle": [Function],
                  },
                  Object {
                    "onProcessStyle": [Function],
                  },
                  Object {
                    "onChangeValue": [Function],
                    "onProcessStyle": [Function],
                  },
                  Object {
                    "onProcessStyle": [Function],
                  },
                  Object {
                    "onChangeValue": [Function],
                    "onProcessRule": [Function],
                    "onProcessStyle": [Function],
                  },
                  Object {
                    "onProcessStyle": [Function],
                  },
                ],
              },
              "plugins": PluginsRegistry {
                "hooks": Object {
                  "onChangeValue": Array [
                    [Function],
                    [Function],
                  ],
                  "onCreateRule": Array [
                    [Function],
                    [Function],
                    [Function],
                    [Function],
                    [Function],
                    [Function],
                    [Function],
                    [Function],
                    [Function],
                    [Function],
                  ],
                  "onProcessRule": Array [
                    [Function],
                    [Function],
                  ],
                  "onProcessSheet": Array [],
                  "onProcessStyle": Array [
                    [Function],
                    [Function],
                    [Function],
                    [Function],
                    [Function],
                    [Function],
                    [Function],
                    [Function],
                  ],
                },
              },
              "version": "8.1.0",
            },
            "meta": "Jss(_ContentContainer), Unthemed, Static",
            "parent": [Circular],
            "sheet": [Circular],
          },
          "renderer": DomRenderer {
            "element": <style
              data-jss=""
              data-meta="Jss(_ContentContainer), Unthemed, Static"
              type="text/css"
            >
              
.app-0-0 {
  margin: auto;
  max-width: 1200px;
}

            </style>,
            "getSelector": [Function],
            "getStyle": [Function],
            "hasInsertedRules": false,
            "setSelector": [Function],
            "setStyle": [Function],
            "sheet": [Circular],
          },
          "rules": RuleList {
            "classes": Object {
              "app": "app-0-0",
            },
            "index": Array [
              Object {
                "margin": "auto",
                "max-width": "1200px",
              },
            ],
            "map": Object {
              ".app-0-0": Object {
                "margin": "auto",
                "max-width": "1200px",
              },
              "app": Object {
                "margin": "auto",
                "max-width": "1200px",
              },
            },
            "options": Object {
              "Renderer": [Function],
              "classes": Object {
                "app": "app-0-0",
              },
              "generateClassName": [Function],
              "index": -100000,
              "insertionPoint": undefined,
              "jss": Jss {
                "generateClassName": [Function],
                "options": Object {
                  "Renderer": [Function],
                  "createGenerateClassName": [Function],
                  "plugins": Array [
                    Object {
                      "onCreateRule": [Function],
                      "onProcessRule": [Function],
                    },
                    Object {
                      "onProcessStyle": [Function],
                    },
                    Object {
                      "onProcessStyle": [Function],
                    },
                    Object {
                      "onProcessStyle": [Function],
                    },
                    Object {
                      "onProcessStyle": [Function],
                    },
                    Object {
                      "onChangeValue": [Function],
                      "onProcessStyle": [Function],
                    },
                    Object {
                      "onProcessStyle": [Function],
                    },
                    Object {
                      "onChangeValue": [Function],
                      "onProcessRule": [Function],
                      "onProcessStyle": [Function],
                    },
                    Object {
                      "onProcessStyle": [Function],
                    },
                  ],
                },
                "plugins": PluginsRegistry {
                  "hooks": Object {
                    "onChangeValue": Array [
                      [Function],
                      [Function],
                    ],
                    "onCreateRule": Array [
                      [Function],
                      [Function],
                      [Function],
                      [Function],
                      [Function],
                      [Function],
                      [Function],
                      [Function],
                      [Function],
                      [Function],
                    ],
                    "onProcessRule": Array [
                      [Function],
                      [Function],
                    ],
                    "onProcessSheet": Array [],
                    "onProcessStyle": Array [
                      [Function],
                      [Function],
                      [Function],
                      [Function],
                      [Function],
                      [Function],
                      [Function],
                      [Function],
                    ],
                  },
                },
                "version": "8.1.0",
              },
              "meta": "Jss(_ContentContainer), Unthemed, Static",
              "parent": [Circular],
              "sheet": [Circular],
            },
            "raw": Object {
              "app": Object {
                "margin": "auto",
                "maxWidth": "1200px",
              },
            },
          },
        }
      }
    />
  </_ContentContainer>
</Jss(_ContentContainer)>
`;
